### DISCUSSION TITLE
Does MSI Installer support scope user?

### DISCUSSION THREAD

**Discussion by BrandonWanHuanSheng** [microsoft/winget-pkgs#103877] [Q&A] [answered]
It look like I am trying to resolves an issues. But suddenly goes validation error on the pipelines. It look like it did not work on Azure Pipelines.  That PR is 103854. Does MSI Installer support scope user?

**Comment by Trenly** [comment, marked as answer]
It depends on the installer. If it is an MSI built with Wix, often times a user scoped install will still write its entries to the machine scoped registry incorrectly. This is a bug with the wix installer itself, and makes it difficult for winget to reason about them effectively. If it is an MSI that isn’t built with wix, it depends on how the publisher created the assembly. Some publishers will specify that their msi can only install to a machine context, others will specify that it can only install to a user context. The most common that I've seen is where if running elevated it will install to machine and if non-elevated it will install to user (think scope: machine & elevationRequired; and scope:user & elevationProhibited). Basically, the only way to know for sure is to see if the publisher has any information on how they install, to test how it installs elevated/non-elevated, or to see if there are any cli switches provided.

TLDR - Yes they can support user scope, but only if the publisher supports it

**Reply by filips123** [reply]
It's WiX installer, and its source is [here](https://github.com/filips123/PWAsForFirefox/blob/main/native/packages/wix/main.wxs). It does support both per-user and per-machine (default) installation, as it offers a dialog during the installation to choose the scope. More details are also [here](https://github.com/microsoft/winget-pkgs/issues/103815#issuecomment-1528751661).

I thought that the user scope could be selected my passing `MSIINSTALLPERUSER=1` property to the installer. This also seems to work when manually running the installer with `msiexec`. However, as @BrandonWanHuanSheng already said, when adding that property, the PR validation failed (#103854).
